### PR TITLE
Fix compiling with THREADS=ON

### LINKED ISSUE #7245: Compiling without threads broken
https://github.com/dealii/dealii/pull/7235 implies that we always use `std::mutex` and `std::lock_guard` even when compiling with `DEAL_II_WITH_THREADS=OFF` we either need to guard all of these cases or always compile with `THREADS` support.

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes #7245.

### CODE REVIEW

File: include/deal.II/lac/swappable_vector.templates.h [hunk 1]
```diff
@@ -75,10 +75,12 @@ SwappableVector<number>::operator=(const SwappableVector<number> &v)
   if (filename != "")
     kill_file();
 
-  // if in MT mode, block all other
-  // operations. if not in MT mode,
-  // this is a no-op
+    // if in MT mode, block all other
+    // operations. if not in MT mode,
+    // this is a no-op
+#ifdef DEAL_II_WITH_THREADS
```
Comment (drwells) [inline include/deal.II/lac/swappable_vector.templates.h]: This is some strange indentation.

**Comment by bangerth** [comment]
Hm, I wished this hadn't been committed. This is the wrong approach, it completely obfuscates a lot of code. Why do we need this?

**Comment by bangerth** [comment]
Is the subject of the PR wrong? Should it have said if "TREADS=OFF"?

**Comment by drwells** [comment]
> Is the subject of the PR wrong?

Yes.

> Why do we need this?

Its ugly, but I think its better to have a blip in the history then break compilation. We need this since we cannot call, e.g., `std::lock_guard(this->mutex)` when `this->mutex` is a `DummyThreadMutex`. This is the disadvantage of not using our own wrapper classes.

**Comment by bangerth** [comment]
I was confused temporarily by the subject. I realize now why it was necessary.

**Comment by masterleinad** [comment]
> I was confused temporarily by the subject. I realize now why it was necessary.

Sorry for the confusion! I was rushing to fix compilation.

**Pull request merged**